backend/dataall/modules/notebooks/db/repositories.py

```diff
@@ -0,0 +1,53 @@
+"""
+DAO layer that encapsulates the logic and interaction with the database for notebooks
+Provides the API to retrieve / update / delete notebooks
+"""
+from sqlalchemy import or_
+from sqlalchemy.orm import Query
+
+from dataall.db import paginate
+from dataall.modules.notebooks.db.models import SagemakerNotebook
+
+
+class NotebookRepository:
```

Comment:
I would introduce some hierarchy to enforce naming convention for CRUD operations that needs to be implemented like for instance Spring Data module does.

Reply:
I actually like the idea of creating some convension, but I think Spring Data is a different story here: it uses `bytebuddy` to create an implementation of a repository based upon your interface. If you don't provide the method name in specific way, the spring boot magic won't work (e.g. `findNotebooksByInstanceTypeAndRegion`). This **enforce** you to write a code in the specific way.

We can create a hierarchy trying to achive the same, e.g. a base class with a generic methods like `save/delete/find`. All repos will extend from a the base repo and can re-use.this method. Hovewer, I still have a concern that when you need a specific method nothing will stop  you from creating something like that `remove_something_by_key` instead of `delete_by_key`, `create_object_with_default` instead of `save_with_default`. So we can end up with a useless base class at the end that brings only complexity. 🤔 
But please let me know what you think.

P.S. I believe if we truly want to create conventions we should create a code style guide where we can describe rules how the code should be written

Reply:
Is there a way to generate a boilerplate module? Like introducing the module name "notebooks" and have some sort of command line that generates a directory of gql, services, db code with the input given. Like a quickstart on developing modules (maybe not for a first PR but something to thin about)